### ISSUE TITLE
Infinite Login Loop After Enabling 2FA

### ISSUE THREAD

**Issue by keokilee** [getsentry/sentry#19820]
Just started using Sentry On Premise for our company and set it up successfully. I was able to log in and send invites to my team. After sending the invites, I enabled 2FA for all accounts. Now, when people accept the invite, they see the following:

![sentry-error](https://user-images.githubusercontent.com/191770/86965551-74fcab00-c103-11ea-8591-5731e6e951bd.png)

I also see a member with the same name/email in my members list, but without 2FA enabled. The solution appears to be to remove that duplicate member and resend the invite, although coworkers are telling me that the new invite puts them into a loop when trying to configure 2FA. They then close that window, open the link again, go through the forgot password flow, and successfully get set up.

tl;dr The steps are:

1. Admin invites members to Sentry
2. Enable 2FA for all members
3. Member clicks invite link, can't configure 2FA
4. Admin removes duplicated member and resends invite
5. Member clicks invite link, prompted to configure 2FA, but gets stuck in loop again
6. Member clicks invite link again, logs in, and can configure 2FA.

Apologies for the lack of logs as I'm not sure what logs correspond to what they're seeing.

**Comment by EvanPurkhiser** [comment]
Cool, I can reproduce this :)

Looks like there are a couple of bugs here. I'll see if I can get this fixed up

**Comment by keokilee** [comment]
Thank you! We're still getting used to the product, but appreciate you guys jumping on to fix this so quickly.

**Comment by patschi** [comment]
Today I've updated from `releases/20.10.1` to `20.11.1` and then I've experienced a redirection loop during login after completing 2FA. While digging through the code and researching I've found this issue, that's why I'm just sharing my results here.

It turns out that completely disabling 2FA (server-side) was the workaround for me. Now I'm able to login again.

**Important**: This workaround worked for me. I'm not sure if that is the best way to go, how it behaves in the future nor what happens when a org enforces 2FA (wasn't in my case). So, on your own risk, have a backup, etc.

Essentially, what I did:
```python
$ docker-compose exec web sentry shell
>>> from sentry.models import User, Authenticator
>>> myUser = User.objects.get(username="user@domain.tld")
>>> print Authenticator.objects.user_has_2fa(myUser.id)
True
>>> Authenticator.objects.filter(user=myUser)
<BaseQuerySet [<Authenticator user=u'user@domain.tld' interface='u2f'>, <Authenticator user=u'user@domain.tld' interface='recovery'>, <Authenticator user=u'user@domain.tld' interface='totp'>]>
>>> Authenticator.objects.filter(user=myUser).delete() # <- CRITICAL PART
>>> print Authenticator.objects.user_has_2fa(myUser.id)
False
>>> Authenticator.objects.filter(user=myUser)
<BaseQuerySet []>
```

**Comment by BYK** [comment]
@patschi do you have any logs from this redirect loop that you can share in a new issue? As much as your workaround is nice, we'd love to find the root cause and fix it for everyone if possible.
